backend/apps/predictions/github_service.py

```
@@ -0,0 +1,71 @@
+        # Fallback simulated data for offline/test environments
+        return {
+            "pr_number": pr_number,
+            "title": f"Feature Implementation #{pr_number}",
+            "author": "contributor",
+            "additions": 150 + (pr_number * 10) % 500,
+            "deletions": 30 + (pr_number * 5) % 200,
+            "changed_files": 4 + (pr_number % 8),
+            "requested_reviewers": ["maintainer-1"],
+        }
+
+    def sync_pr_metric(self, pr_data: dict, assigned_reviewer_username: str = None) -> PullRequestMetric:
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)